test/system-tests/request_handler.cpp

```diff
@@ -240,9 +221,34 @@ void RequestHandler::on_inject_headers(const httplib::Request& req,
   res.set_content(response_json.dump(), "application/json");
 }
 
+void RequestHandler::on_extract_headers(const httplib::Request& req,
+                                        httplib::Response& res) {
+  const auto request_json = nlohmann::json::parse(req.body);
+  auto http_headers = utils::get_if_exists<nlohmann::json::array_t>(
+      request_json, "http_headers");
+  if (!http_headers) {
+    VALIDATION_ERROR(res, "on_extract_headers: missing `http_headers` field.");
+  }
+
+  datadog::tracing::SpanConfig span_cfg;
+  auto span =
+      tracer_.extract_span(utils::HeaderReader(*http_headers), span_cfg);
+  if (auto error = span.if_error()) {
+    VALIDATION_ERROR(res, error->with_prefix("on_extract_headers: ").message);
+  }
+
+  const auto response_body = nlohmann::json{
+      {"span_id", span->id()},
```

Comment:
Here we want to return the span_id extracted from the distributed tracing headers not the span id of the newly created span.

```suggestion
      {"span_id", span->parent_id()},
```